tests/integration_tests/data_steward/gcloud/gcs_client_test.py

```diff
@@ -0,0 +1,56 @@
+"""
+Test the Google Cloud Storage Client and associated helper functions
+"""
+# Python stl imports
+import os
+import unittest
+
+# Project imports
+from gcloud.gcs import StorageClient
+
+# Third-party imports
+
+
+class GcsClientTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        print('**************************************************************')
+        print(cls.__name__)
+        print('**************************************************************')
+
+    def setUp(self):
+        self.client = StorageClient()
+        self.bucket_name: str = os.environ.get('BUCKET_NAME_FAKE')
+        self.prefix: str = 'prefix'
+        self.data: bytes = b'bytes'
+
+        self.sub_prefixes: tuple = (f'{self.prefix}/a', f'{self.prefix}/b',
+                                    f'{self.prefix}/c', f'{self.prefix}/d')
+
+    def test_list_sub_prefixes(self):
+        self._empty_bucket()
+        self._stage_bucket()
+
+        items = self.client.list_sub_prefixes(self.bucket_name, self.prefix)
+        print(items)
+
+        self.assertEqual(len(self.sub_prefixes), len(items))
+        for item in items:
+            self.assertIn(item[:-1], self.sub_prefixes)
```

Comment:
nit: One of the most helpful unit test assertions IMO is `assertCountEqual`. It can compare unhashable types and is order agnostic. Despite its name, it verifies that all items in one exist in the other (essentially counts of each item are the same in both).
```suggestion
        items = self.client.list_sub_prefixes(self.bucket_name, self.prefix)

        self.assertCountEqual(items, self.sub_prefixes)
```
This will need minor adjustments like adding `/` to each sub_prefix, so I am not making this a required change.
You can also get rid of the print statement.

Reply:
First: good to know about `assertCountEqual`!  I'll keep that in mind for future tests.

Second: The `print()` statement was removed.  It should not have made it in.

Third:  I considered adding the trailing `/` into the sub_prefixes so that the `self.assertIn()` call wouldn't have the splice `[:-1]`, but ultimately I felt this made the code less readable in the upload section:

`bucket.blob(f'{sub_prefix}obj.txt').upload_from_string(self.data)`

That line without the `/` seperating the filename/blob name looked odd, so I opted for the string splice.

Thanks for the positive feedback!

Reply:
You're right, it does look odd. The `sub_prefix` would then need the splice, but interestingly, it seems to work even with the `/` added without the splice as well.